### PR TITLE
Fix avus with integers

### PR DESCRIPTION
**Pull request by montesmariana**

### CODE REVIEW

File: mango_mdschema/fields.py [hunk 1]
```diff
@@ -823,10 +815,18 @@ def convert(self, value):
                         value,
                     )
                 return []
+            if not all(v in self.choices for v in values):
+                logger.info(
+                    f"Some values in '{self.name}' were not allowed and are discarded. Allowed values: {self.choices}."
```
Comment (rmoreas) [inline mango_mdschema/fields.py]: %-formatting should be used for logging. See https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/logging-fstring-interpolation.html

File: mango_mdschema/fields.py [hunk 2]
```diff
@@ -823,10 +815,18 @@ def convert(self, value):
...
+                )
             return [v for v in values if v in self.choices]
         # if we have only one value
-        value = str(value) if value is not None else None
-        return value if value in self.choices else None
+        if value not in self.choices and not self.required:
+            logger.info(
+                f"'{self.name} must be one of the following values: '{self.choices}', got '{value}'. It was discarded."
```
Comment (rmoreas) [inline mango_mdschema/fields.py]: %-formatting should be used for logging. See https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/logging-fstring-interpolation.html

**Comment by rmoreas** [comment]
Besides remarks about %-formatting for logging, it looks good to me.

Adding a test  for the creation of the log message, would make it perfect ;-)

Can be done with assetLog context manager: https://docs.python.org/3.9/library/unittest.html#unittest.TestCase.assertLogs

**Review by rmoreas — APPROVED** [review]

**Comment by montesmariana** [comment]
> Besides remarks about %-formatting for logging, it looks good to me.
> 
> Adding a test for the creation of the log message, would make it perfect ;-)
> 
> Can be done with assetLog context manager: https://docs.python.org/3.9/library/unittest.html#unittest.TestCase.assertLogs

Sounds great but I'll create an issue for it and do it in a later PR, because I need to get this out soon. Thank you!!

**Pull request merged**